mirrord/layer/src/socket.rs

```diff
@@ -26,11 +26,18 @@ pub(crate) mod ops;
 pub(crate) static SOCKETS: LazyLock<Mutex<HashMap<RawFd, Arc<UserSocket>>>> =
     LazyLock::new(|| Mutex::new(HashMap::new()));
 
-/// Holds the connections that are waiting to be fully realized.
+/// Holds the connections that have yet to be [`accept`](ops::accept)ed.
+///
+/// ## Details
 ///
 /// The connections here are added by
-/// [`TcpHandler::create_local_stream`](crate::tcp::TcpHandler::create_local_stream) and are dealt
-/// with by [`ops::accept`].
+/// [`TcpHandler::create_local_stream`](crate::tcp::TcpHandler::create_local_stream) when the agent
+/// sends us a [`NewTcpConnection`](mirrord_protocol::tcp::NewTcpConnection).
+///
+/// And they become part of the [`UserSocket`]'s [`SocketState`] when [`ops::accept`] is called.
+///
+/// Finally, we remove a [`ConnectionQueue`] when the socket's `fd` is closed in
```

Comment:
The new documentation here looks good! This line is confusing though, since there is only 1 `ConnectionQueue` and we don't remove it.
```suggestion
/// Finally, we remove a socket's queue when the socket's `fd` is closed in
```